packages/fast-components-react-base/src/listbox/listbox.tsx

```diff
@@ -0,0 +1,314 @@
+import * as ReactDOM from "react-dom";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import { ListboxClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import {
+    ListboxHandledProps,
+    ListboxProps,
+    ListboxUnhandledProps,
+} from "./listbox.props";
+import * as React from "react";
+import { KeyCodes } from "@microsoft/fast-web-utilities";
+import { get, inRange, invert } from "lodash-es";
+import { canUseDOM } from "exenv-es6";
+
+export interface ListboxState {
+    /**
+     * The index of the focusable child
+     */
+    focusIndex: number;
+}
+
+export interface TypeAheadDataItem {
+    index: number;
+    compareString: string;
+}
+
+class Listbox extends Foundation<
+    ListboxHandledProps,
+    ListboxUnhandledProps,
+    ListboxState
+> {
+    public static displayName: string = "Listbox";
+
+    protected handledProps: HandledProps<ListboxHandledProps> = {
+        children: void 0,
+        managedClasses: void 0,
+        typeAheadPropName: void 0,
```

Comment:
`Prop` feels weird within the context of the prop contract; is there a better name that doesn't contain "prop"?

Reply:
I will add the activeDescendant bit, seems straightforward.  As for multi-select -  this component does not know about selection at all in the current implementation, it essentially is a focus manager for a menu.  Managing selection state is the job of whatever component owns the listbox, most likely a Select component, and the options themselves.  (And yes, I have multi-select working in my working branch)

Reply:
I think the difficult thing here is that in the listbox is an accessible and functional pattern itself; not a partial pattern - and part of the pattern is that it includes the selection as part of its functional spec (https://www.w3.org/TR/wai-aria-practices-1.1/#Listbox). I think it would be odd for us to implement this partially when an expectation of a user would likely be that this supports the functionality of a listbox

Reply:
Discussed name change, we decided to with Jane's suggestion of "typeAheadPropertyKey".  That and activeDescendant changes queued for next iteration.